### ISSUE TITLE
Shouldn't non available resources thrown an IOException? 

### ISSUE THREAD

**Issue by nkutsche** [cmarchand/cp-protocol#1]
Hi,

I'm trying this out in combination with the [catalogBuilder-maven-plugin](https://github.com/cmarchand/maven-catalogBuilder-plugin). I have now problems by testing whether a resource is available or not. In XSLT I'm using e.g. `doc-available('cp:/path/to/some/optional/resource.xml')`. But I got a NullPointerException, if the resource is not available. 
I think this could be fixed, if the `CpUrlConnection` throws an `IOException` when it gets `null` from the `path` ([here](https://github.com/cmarchand/cp-protocol/blob/master/src/main/java/top/marchand/xml/protocols/cp/Handler.java#L63)), shouldn't it?

I did a short test and this seems to work for my case:

```java

        @Override
        public InputStream getInputStream() throws IOException {
            ClassLoader cl = Handler.CP_THREAD_LOCAL.get();
            if(cl==null) cl = Thread.currentThread().getContextClassLoader();
            InputStream inputStream = cl.getResourceAsStream(path);
            if(inputStream==null) 
                throw new IOException("Could not find resource " + path);
            return cl.getResourceAsStream(path);
        }
```

Best Reagrds, Nico

**Comment by cmarchand** [comment]
I think you are right. I have no such test case.
Let's ask to @mricaud and @axelCourt if they have such a test-case, and their opinion...

**Comment by mricaud** [comment]
Sorry I don't have such a use-case, because I prevent using cp:/ url directly in my XSLT. 
I rather use a dependency:/ protocol in my code, which might (or not) be resolved at run-time to a cp:/ url thanks to the XML Catalog. Doing this let the user of the XSL choose how to resolve external urls within its own catalog.

**Comment by nkutsche** [comment]
Hi, thanks for your responses! Actually I also don't use `cp:/` directly in XSLT, but use a catalog. My example was just for simplification.
I think the special of my use-case is, that I want to check whether a resource is available in my classpath or not.
And this doesn't work, because any request that is directed to that protocol ends up with a nullpointer if it does not lead to an existing resource. And I think the correct behavior would be to throw an IOException, wouldn't it?

**Comment by cmarchand** [comment]
It would, very probably. I have to check the javadoc, but I do agree with you.
If there is no following in the next week, plus send me a reminder.

**Comment by cmarchand** [comment]
Javadoc says :

> Throws:
> IOException - if an I/O error occurs while creating the input stream.
> UnknownServiceException - if the protocol does not support input.

And in this case, we weren't able to create the InputStream. So, yes, it should throw an IOException.

**Comment by cmarchand** [comment]
I have a problem : I've upgraded my JDK to openjdk version "11.0.4" 2019-07-16, and now I have errors on build, with javadoc.
```
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in http://docs.oracle.com/javase/7/docs/api/ are in the unnamed module.
```
If someone knows how to fix this, any help will be much appreciated !

**Comment by cmarchand** [comment]
Problem solved...

**Comment by cmarchand** [comment]
Release 2.2 published. Just wait for Nexus synchronization...

**Comment by nkutsche** [comment]
I'm a bit confused. I was just faced with the same issue and the version 2.2 didn't helped.

It seems a) there was no fix for this issue in 2.2 and b) I didn't re-checked it after the release. I can't remember where I was faced this issue the first time - maybe I implemented a workaround and just forgot to try to replace it by using 2.2.

I'll create a pull request...
